core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,8 +389,18 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
```

Comment:
We should check view first, before checking the table 

https://github.com/trinodb/trino/blob/371fc01e1ae67359f9f872602bbb30f2bc87265e/core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java#L1228-L1233